torch/_inductor/codegen/cpp_wrapper_cpu.py

```diff
@@ -2349,13 +2352,15 @@ def extract_output_name(
             raise AssertionError(f"Unexpected output: {type(out)}")
 
         if isinstance(op_overload, torch._ops.HigherOrderOperator):
-            assert isinstance(
-                op_overload, torch._higher_order_ops.torchbind.CallTorchBind
-            ), type(op_overload)
-            assert len(raw_args) > 1
-            obj = raw_args[0]
-            method = raw_args[1]
-            return_schema = op_overload.schema(obj, method).returns
+            if isinstance(op_overload, torch._higher_order_ops.torchbind.CallTorchBind):
+                assert len(raw_args) > 1
+                obj = raw_args[0]
+                method = raw_args[1]
+                return_schema = op_overload.schema(obj, method).returns
+            else:
+                # For non-CallTorchBind HOPs (like print), we don't need schema for C++ wrapper
+                # since they're handled specially (e.g., printf for print)
+                return_schema = []
```

Comment:
since print is the only one that should use this path currently, could we explicitly assert here that the op is  `torch.ops.higher_order.print`?

Reply:
That's a great point. Initially (before print), this part seems to only support ``torch._higher_order_ops.torchbind.CallTorchBind``. I'll specify the error handling if it's neither torchbind nor print.